csrc/scheduler/matmul.cpp

```diff
@@ -614,47 +617,96 @@ void scheduleOutputTensor(
       {c},
       {ParallelType::BIDx, ParallelType::BIDy, ParallelType::BIDz});
 }
+
+//! Assign parallelization modes to epilogue inputs. The returned pair of
+//!  collections contains:
+//!   - 1st: TVs that can have all parallelization modes propagated,
+//!   - 2nd: TVs that can have all but vectorization parallel modes
+//!     propagated,
+std::pair<std::vector<TensorView*>, std::vector<TensorView*>>
+partitionEpilogueInputsByParallelizationModes(
+    TensorView* reference_tv,
+    const std::vector<TensorView*>& epilogue_inputs) {
+  const auto& tvs_with_vec = scheduler_utils::getInputsOutputsWithInnerDim(
+      reference_tv, true /*inner_only*/, true /*vectorize_pass*/);
+
+  std::vector<TensorView*> epilogue_input_tvs_with_vec = {};
```

Comment:
For this use case I prefer to rely on the initial size of vector set during construction, and only if there is not enough space, let the STL implementation of vector handle the resize (usually it doubles the reserved size if limit is reached).

Details:

The reason is that `getInputsOutputsWithInnerDim(..)` will look for vectorizable TVs in fusion's inputs and outputs, while for us the scope is limited to fusion inputs with `INPUT_C`  roles, that is additional inputs that are used in epilogue. I explicitly avoid MMA inputs, as they are processed separately.

For first example, for vanilla matmul (`D = A x B`) this function can return up to 2 TVs (`A` and `B`, `D` is our reference here), none of them can be considered so we will end up with empty `epilogue_input_tvs_with_vec`.

For second, more advanced, example with fusion `D = AxB + beta * C`, the `getInputsOutputsWithInnerDim(..)` call may return 3 TVs (`A`, `B` and `C`) where only `C` will be further considered.

With reserve we will have most likely unneeded memory operations (allocation, copy and deallocation of previous buffer). If we were to deal with `TensorViews` objects directly then it might be worth considering, but here we operate on pointers.

Reply:
I know it is not a big deal in this case since the number of added elements is very small. I am just a little confused by `With reserve we will have most likely unneeded memory operations` (I think `reserve` is used to avoid these) and `I prefer to rely on the initial size of vector set during construction` (I think the initial size is 0).

Here is my logic:
Initially, without using reserve, the vector is created with both size and capacity set to 0. As elements are added to the vector, its capacity is automatically increased, typically doubling each time: from 1 to 2, then 4, 8, and so on. Each increase in capacity may involve additional memory operations, as the existing elements need to be copied to a new location, and the old memory needs to be deallocated. Is there any hole?

Reply:
Thanks @liqiangxl for reply and sharing the details. I double checked what standard says about initial size of allocated space for `std::vector` and this is not defined, but most of the current implementations does start with `0`. Then your logic has no holes there. What I wrote is based on experiments I've done in the past, but this looks like changed. I will update the implementation to follow your advice.

Thanks!